pkg/sources/github/github.go

```diff
@@ -790,34 +792,39 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 			s.setScanOptions(s.conn.Base, s.conn.Head)
 
 			repoSize := s.repoSizes.getRepo(repoURL)
-			logger.V(2).Info(fmt.Sprintf("scanning repo %d/%d", i, len(s.repos)), "repo_size_kb", repoSize)
+			logger.V(2).Info("scanning repo", "repo_size_kb", repoSize)
 
 			now := time.Now()
-			defer func(start time.Time) {
-				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scanned, len(s.repos)), "repo_size", repoSize, "duration_seconds", time.Since(start).Seconds())
-			}(now)
+			scanFailed := false
+			defer func(start time.Time, failed *bool) {
+				if *failed {
+					return
+				}
+				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", time.Since(start).Seconds())
+			}(now, &scanFailed)
 
 			if err = s.git.ScanRepo(ctx, repo, path, s.scanOptions, sources.ChanReporter{Ch: chunksChan}); err != nil {
 				scanErrs.Add(fmt.Errorf("error scanning repo %s: %w", repoURL, err))
+				scanFailed = true
 				return nil
 			}
 
 			githubReposScanned.WithLabelValues(s.name).Inc()
 
 			if err = s.scanComments(ctx, repoURL, chunksChan); err != nil {
 				scanErrs.Add(fmt.Errorf("error scanning comments in repo %s: %w", repoURL, err))
+				scanFailed = true
 				return nil
 			}
 
-			atomic.AddUint64(&scanned, 1)
-
+			atomic.AddUint64(&scannedCount, 1)
 			return nil
 		})
 	}
 
 	_ = s.jobPool.Wait()
 	if scanErrs.Count() > 0 {
-		s.log.V(2).Info("Errors encountered while scanning", "error-count", scanErrs.Count(), "errors", scanErrs)
+		s.log.V(0).Info("failed to scan some repositories", "error_count", scanErrs.Count(), "errors", scanErrs)
```

Comment:
I think it's important to have this at the default logging level. Otherwise, people may not realize that a repository was not scanned.

Is there a reason we wait until the end instead of logging failures as they happen?

Reply:
I'm not sure if there is a good reason for it, but I will merge as-is because this is an improvement and not changing that functionality.